### PR TITLE
[Rule Tuning] Update EQL rules for 7.10

### LINKED ISSUE #396: [Rule Tuning] Add case-insensitive operator to EQL queries
## Description
We should switch all case-insensitive `==` to the EQL `:` operator.

For unit tests to pass, we'll need the Python module 0.9.5 update to add an [ES syntax toggle](https://github.com/endgameinc/eql/pull/410)

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
dependent on endgameinc/eql/pull/41
resolves #396

## Summary
Updates EQL rules to reflect changes in EQL implementation and syntax. Rules were also tuned to adapt to missing fields in respective data sources (or the indexes were removed). EQL searches are strict and will fail if a field does not exist in the mapping

### CODE REVIEW

File: rules/windows/defense_evasion_suspicious_wmi_script.toml [hunk 1]
```diff
@@ -22,19 +22,11 @@ tags = ["Elastic", "Windows"]
...
-
-
 sequence by process.entity_id with maxspan=2m
 [process where event.type in ("start", "process_started") and
-   (process.name == "wmic.exe" or process.pe.original_file_name == "wmic.exe" or
-      winlog.event_data.OriginalFileName == "wmic.exe") and
-   wildcard(process.args, "format*:*", "/format*:*", "*-format*:*") and
-   not process.args in ("/format:table", "/format:table") or wildcard(process.args, "format*:*")]
+   (process.name == "WMIC.exe" or process.pe.original_file_name == "wmic.exe") and
+   (process.args:"format*:*" or process.args:"/format*:*" or process.args:"*-format*:*") and
```
Comment (rw-access) [inline rules/windows/defense_evasion_suspicious_wmi_script.toml:28]:
```suggestion
   wildcard(process.args, "format*:*", "/format*:*", "*-format*:*") and
```

File: rules/windows/defense_evasion_suspicious_wmi_script.toml [hunk 2]
```diff
@@ -22,19 +22,11 @@ tags = ["Elastic", "Windows"]
...
+   not process.command_line:"* /format:table *"]
```
Comment (rw-access) [inline rules/windows/defense_evasion_suspicious_wmi_script.toml:29]:
```suggestion
   not wildcard(process.command_line, "* /format:table *")]
```

File: rules/windows/lateral_movement_cmd_service.toml [hunk 3]
```diff
@@ -22,15 +22,14 @@ tags = ["Elastic", "Windows"]
...
-     (process.name == "sc.exe" or process.pe.original_file_name == "sc.exe" or
-        winlog.event_data.OriginalFileName == "sc.exe") and
-     wildcard(process.args, "\\\\*") and wildcard(process.args, "binPath*", "binpath*") and
-     process.args in ("create", "config", "failure", "start")]
+     (process.name == "sc.exe" or process.pe.original_file_name == "sc.exe") and
+                                                   /* case insensitive */
+      process.command_line:"* \\\\* binPath=*" and process.args:"create" or
+                                                   process.args:"config" or
+                                                   process.args:"failure" or
+                                                   process.args:"start"]
```
Comment (rw-access) [inline rules/windows/lateral_movement_cmd_service.toml:32]:
```suggestion
      wildcard(process.args, "\\\\*") and wildcard(process.args, "binPath=*") and (
          process.args : "create" or
          process.args : "config" or
          process.args : "failure" or
          process.args : "start")]
```

File: rules/windows/defense_evasion_rundll32_no_arguments.toml [hunk 4]
```diff
@@ -22,21 +22,17 @@ tags = ["Elastic", "Windows"]
...
 
      /* zero arguments excluding the binary itself (and accounting for when the binary may not be logged in args) */
      ((process.args == "rundll32.exe" and process.args_count == 1) or
       (process.args != "rundll32.exe" and process.args_count == 0))
 
   ] by process.entity_id
   [process where event.type in ("start", "process_started") and
-     (process.name == "rundll32.exe" or process.pe.original_file_name == "rundll32.exe" or
-        winlog.event_data.OriginalFileName == "rundll32.exe")
+     (process.name == "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE")
```
Comment (rw-access) [inline rules/windows/defense_evasion_rundll32_no_arguments.toml:35]:
```suggestion
     (process.name : "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE")
```

File: rules/windows/defense_evasion_rundll32_no_arguments.toml [hunk 5]
```diff
@@ -22,21 +22,17 @@ tags = ["Elastic", "Windows"]
 type = "eql"
 
 query = '''
-/* replace winlog.event_data.OriginalFileName with process.pe.original_file_name once field is available in winlogbeat */
-
 sequence with maxspan=1h
   [process where event.type in ("start", "process_started") and
-     (process.name == "rundll32.exe" or process.pe.original_file_name == "rundll32.exe" or
-        winlog.event_data.OriginalFileName == "rundll32.exe") and
+     (process.name == "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE") and
```
Comment (rw-access) [inline rules/windows/defense_evasion_rundll32_no_arguments.toml:27]:
```suggestion
     (process.name : "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE") and
```

File: rules/windows/defense_evasion_rundll32_no_arguments.toml [hunk 6]
```diff
@@ -22,21 +22,17 @@ tags = ["Elastic", "Windows"]
...
 
      /* zero arguments excluding the binary itself (and accounting for when the binary may not be logged in args) */
      ((process.args == "rundll32.exe" and process.args_count == 1) or
       (process.args != "rundll32.exe" and process.args_count == 0))
```
Comment (rw-access) [inline rules/windows/defense_evasion_rundll32_no_arguments.toml:31]: `process.args_count` will be 1 at a minimum.
like argv in c/python, the first argument is (typically) the path for rundll32.

File: rules/windows/lateral_movement_cmd_service.toml [hunk 7]
```diff
@@ -22,15 +22,14 @@ tags = ["Elastic", "Windows"]
...
   [network where event.type == "connection" and process.name == "sc.exe" and destination.address != "127.0.0.1"]
```
Comment (rw-access) [inline rules/windows/lateral_movement_cmd_service.toml:33]:
```suggestion
  [network where event.type == "connection" and process.name : "sc.exe" and destination.address != "127.0.0.1"]
```

File: rules/windows/defense_evasion_installutil_beacon.toml [hunk 8]
```diff
@@ -22,7 +22,7 @@ tags = ["Elastic", "Windows"]
 type = "eql"
 
 query = '''
-/* this can be done without a sequence however, this does include more info on the process */
+/* the benefit of doing this as an eql sequence vs kql is this will limit to alerting only on the first network connection */
 
 sequence by process.entity_id
   [process where event.type in ("start", "process_started") and process.name == "installutil.exe"]
```
Comment (rw-access) [inline rules/windows/defense_evasion_installutil_beacon.toml:28]:
```suggestion
  [process where event.type in ("start", "process_started") and process.name : "installutil.exe"]
```

File: rules/windows/defense_evasion_reg_beacon.toml [hunk 9]
```diff
@@ -24,7 +24,7 @@ type = "eql"
 query = '''
 sequence by process.entity_id
   [process where event.type in ("start", "process_started") and
-     process.name in ("regasm.exe", "regsvcs.exe", "regsvr32.exe")]
+     process.name in ("RegAsm.exe", "regsvcs.exe", "regsvr32.exe")]
```
Comment (rw-access) [inline rules/windows/defense_evasion_reg_beacon.toml:27]: `process.name : x or ...`

File: rules/windows/defense_evasion_rundll32_no_arguments.toml (hunk 6 above)
Comment (rw-access) [reply rules/windows/defense_evasion_rundll32_no_arguments.toml:31]: and if args_count _was_ zero, this `!=` would always be true

File: rules/windows/defense_evasion_suspicious_wmi_script.toml [hunk 10]
```diff
@@ -22,19 +22,11 @@ tags = ["Elastic", "Windows"]
...
-*/
-
-
 sequence by process.entity_id with maxspan=2m
 [process where event.type in ("start", "process_started") and
-   (process.name == "wmic.exe" or process.pe.original_file_name == "wmic.exe" or
-      winlog.event_data.OriginalFileName == "wmic.exe") and
-   wildcard(process.args, "format*:*", "/format*:*", "*-format*:*") and
-   not process.args in ("/format:table", "/format:table") or wildcard(process.args, "format*:*")]
+   (process.name == "WMIC.exe" or process.pe.original_file_name == "wmic.exe") and
```
Comment (rw-access) [inline rules/windows/defense_evasion_suspicious_wmi_script.toml:27]:
```suggestion
   (process.name : "WMIC.exe" or process.pe.original_file_name == "wmic.exe") and
```

File: rules/windows/execution_ms_office_written_file.toml [hunk 11]
```diff
@@ -24,14 +24,14 @@ type = "eql"
 query = '''
 sequence with maxspan=2h
   [file where event.type != "delete" and file.extension == "exe" and
-     process.name in ("winword.exe",
-                      "excel.exe",
-                      "outlook.exe",
-                      "powerpnt.exe",
+     process.name in ("WINWORD.EXE",
```
Comment (rw-access) [inline rules/windows/execution_ms_office_written_file.toml:27]: an `x : ... or x : ... or ...` is the only way to do a case-insensitive `in` for 7.10.

if we don't have time to be complete everywhere and just need to get this through, that's okay. we can pick it up when we have cycles (and aren't both moving)

File: rules/windows/defense_evasion_rundll32_sequence.toml [hunk 12]
```diff
@@ -22,20 +22,16 @@ tags = ["Elastic", "Windows"]
 type = "eql"
 
 query = '''
-/* replace winlog.event_data.OriginalFileName with process.pe.original_file_name once field is available in winlogbeat */
-
 sequence by process.entity_id with maxspan=2h
   [process where event.type in ("start", "process_started") and
-     (process.name == "rundll32.exe" or process.pe.original_file_name == "rundll32.exe" or
-        winlog.event_data.OriginalFileName == "rundll32.exe") and
+     (process.name == "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE") and
```
Comment (brokensound77) [inline rules/windows/defense_evasion_rundll32_sequence.toml:27]:
```suggestion
     (process.name ; "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE") and
```

File: rules/windows/defense_evasion_rundll32_sequence.toml [hunk 13]
```diff
@@ -22,20 +22,16 @@ tags = ["Elastic", "Windows"]
...
 
      /* zero arguments excluding the binary itself (and accounting for when the binary may not be logged in args) */
      ((process.args == "rundll32.exe" and process.args_count == 1) or
       (process.args != "rundll32.exe" and process.args_count == 0))]
 
   [network where event.type == "connection" and
-     (process.name == "rundll32.exe" or process.pe.original_file_name == "rundll32.exe" or
-        winlog.event_data.OriginalFileName == "rundll32.exe")]
+     (process.name == "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE")]
```
Comment (brokensound77) [inline rules/windows/defense_evasion_rundll32_sequence.toml:34]:
```suggestion
     (process.name : "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE")]
```

File: rules/windows/lateral_movement_cmd_service.toml (hunk 3 above)
Comment (brokensound77) [reply rules/windows/lateral_movement_cmd_service.toml:32]: This will `and` the same multi-value field. Does using a method bypass the optimization bug?

Comment (rw-access) [reply rules/windows/lateral_movement_cmd_service.toml:32]: as long as we don't AND with `==` specifically on a field, we should be okay. so an AND on `:` is still safe, or functions

**Review by rw-access — COMMENTED** [review]
definitely getting closer

File: rules/windows/defense_evasion_rundll32_no_arguments.toml (hunk 6 above)
Comment (brokensound77) [reply rules/windows/defense_evasion_rundll32_no_arguments.toml:31]: I have seen the that with certain binaries, the binary can sometimes not be captured as part of the command line, which was the justification for this. I will convert it to normal (your feedback) and if we see the behavior again, revisit this logic

File: rules/windows/lateral_movement_cmd_service.toml (hunk 3 above)
Comment (brokensound77) [reply rules/windows/lateral_movement_cmd_service.toml:32]: thanks for the distinction

File: rules/windows/defense_evasion_msbuild_beacon_sequence.toml [hunk 14]
```diff
@@ -23,8 +23,8 @@ type = "eql"
 
 query = '''
 sequence by process.entity_id
-  [process where event.type in ("start", "process_started") and process.name == "MSBuild.exe"]
-  [network where process.name == "MSBuild.exe" and
+  [process where event.type in ("start", "process_started") and process.name : "MSBuild.exe"]
+  [network where process.name : "MSBuild.exe" and
      not (destination.address == "127.0.0.1" and source.address == "127.0.0.1")]
```
Comment (bm11100) [inline rules/windows/defense_evasion_msbuild_beacon_sequence.toml:28]:
```suggestion
     not (destination.ip == "127.0.0.1" and source.ip == "127.0.0.1")]
```

File: rules/windows/lateral_movement_cmd_service.toml [hunk 15]
```diff
@@ -22,16 +22,16 @@ tags = ["Elastic", "Windows"]
...
-     process.args in ("create", "config", "failure", "start")]
-  [network where event.type == "connection" and process.name == "sc.exe" and destination.address != "127.0.0.1"]
+     (process.name == "sc.exe" or process.pe.original_file_name == "sc.exe") and
+                                                   /* case insensitive */
+      wildcard(process.args, "\\\\*") and wildcard(process.args, "binPath=*", "binpath=*") and 
+      (process.args : "create" or
+       process.args : "config" or
+       process.args : "failure" or
+       process.args : "start")]
+  [network where event.type == "connection" and process.name : "sc.exe" and destination.address != "127.0.0.1"]
```
Comment (bm11100) [inline rules/windows/lateral_movement_cmd_service.toml:34]:
```suggestion
  [network where event.type == "connection" and process.name : "sc.exe" and destination.ip != "127.0.0.1"]
```

File: rules/windows/defense_evasion_msbuild_beacon_sequence.toml (hunk 14 above)
Comment (bm11100) [reply rules/windows/defense_evasion_msbuild_beacon_sequence.toml:28]: address -> ip can be accounted for here or in https://github.com/elastic/detection-rules/pull/400

**Review by bm11100 — APPROVED** [review]
lgtm

**Review by rw-access — APPROVED** [review]
thank you, thank you, thank you!

**Pull request merged**